### PR TITLE
Add tests for release script

### PR DESCRIPTION
**Pull request by sosukesuzuki**
## Description

Adds tests for release script.

This PR adds test for one step (`validate-new-version`). In other PR, I'll add tests for other steps.

## Checklist

- [x] I’ve added tests to confirm my change works.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by sosukesuzuki** [comment]
Why `Dev_Package_Test` failed? It failed on `main` too...

**Comment by fisker** [comment]
This means `yarn.lock` is outdate, and Prettier can't work with some new package.

**Comment by fisker** [comment]
My guess, caused by https://github.com/acornjs/acorn/releases/tag/8.5.0, try regenerate lock file will see the problem.

**Comment by sosukesuzuki** [comment]
@fisker Thank you! I've created PR #11458

File: .github/workflows/release-script-test.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        uses: actions/checkout@v2.3.4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v2.4.0
+        with:
+          version: "14"
+          cache: "yarn"
+
+      - name: Install Dependencies
+        run: cd scripts/release && yarn install --frozen-lockfile
```
Comment (fisker) [inline .github/workflows/release-script-test.yml:28]: Can we use `working-directory`? https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#defaultsrun

Comment (sosukesuzuki) [reply .github/workflows/release-script-test.yml]: 48fc148 , works fine

**Review by fisker — APPROVED** [review]

**Pull request merged**